osu.Game/Overlays/SkinEditor/SkinEditor.cs

```diff
@@ -348,6 +362,7 @@ private void targetChanged(ValueChangedEvent<SkinComponentsContainerLookup?> tar
             changeHandler = new SkinEditorChangeHandler(skinComponentsContainer);
             changeHandler.CanUndo.BindValueChanged(v => undoMenuItem.Action.Disabled = !v.NewValue, true);
             changeHandler.CanRedo.BindValueChanged(v => redoMenuItem.Action.Disabled = !v.NewValue, true);
+            changeHandler.TransactionEnded += () => Mutated = true;
```

Comment:
If you compare with `Editor` which has similar functionality, you'll notice there's already a path to determine the state of the skin using the change handler, namely using `CurrentStateHash` and comparing with the state hash at the point of loading the skin.

In addition, this does not look to be reset to false when the user saves the skin, so the dialog will still appear even though it shouldn't.